google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java

```diff
@@ -186,6 +187,10 @@ public static Builder newBuilder(String topic) {
       return new Builder(topic);
```

Comment:
I think the best idea is to delete the String overload altogether and just use TopicName. TopicName is then responsible for the validation.